### PR TITLE
Outputvar fixes

### PR DESCRIPTION
**Pull request by raghavharness**

### CODE REVIEW

File: pipeline/runtime/runtest.go [hunk 1]
```diff
@@ -67,10 +67,10 @@ func executeRunTestStep(ctx context.Context, engine *engine.Engine, r *api.Start
 		err = collectionErr
 	}
 
-	exportEnvs := fetchExportedVarsFromEnvFile(exportEnvFile, out)
+	exportEnvs, _ := fetchExportedVarsFromEnvFile(exportEnvFile, out)
```
Comment (shubham149) [inline pipeline/runtime/runtest.go:70]: ditto

File: pipeline/runtime/common.go [hunk 2]
```diff
@@ -69,20 +69,21 @@ func isPython(entrypoint []string) bool {
 }
 
 // Fetches variable in env file exported by the step.
-func fetchExportedVarsFromEnvFile(envFile string, out io.Writer) map[string]string {
+func fetchExportedVarsFromEnvFile(envFile string, out io.Writer) (map[string]string, error) {
```
Comment (shubham149) [inline pipeline/runtime/common.go:72]: Why are we returning error if we are not handling in the caller?

File: pipeline/runtime/run.go [hunk 3]
```diff
@@ -49,10 +49,14 @@ func executeRunStep(ctx context.Context, engine *engine.Engine, r *api.StartStep
 		logrus.WithError(rerr).WithField("step", step.Name).Errorln("failed to upload report")
 	}
 
-	exportEnvs := fetchExportedVarsFromEnvFile(exportEnvFile, out)
+	exportEnvs, _ := fetchExportedVarsFromEnvFile(exportEnvFile, out)
```
[the remark this answers is not in the document]
Comment (raghavharness) [reply pipeline/runtime/run.go:52]: Should I just log here ?

File: pipeline/runtime/common.go [hunk 4]
```diff
@@ -69,20 +69,21 @@ func isPython(entrypoint []string) bool {
...
 	if _, err := os.Stat(envFile); errors.Is(err, os.ErrNotExist) {
-		return nil
+		return nil, err
 	}
 
 	env, err := godotenv.Read(envFile)
 	if err != nil {
-		log.WithError(err).WithField("envFile", envFile).Warnln("failed to read exported env file")
-		return nil
+		content, _ := os.ReadFile(envFile)
```
[the remark this answers is not in the document]
Comment (raghavharness) [reply pipeline/runtime/common.go:82]: Hmm but this block is itself inside an error cond. If I override the error in this statement then we won't be able to know the actual error.

[the remark this answers is not in the document]
Comment (shubham149) [reply pipeline/runtime/common.go:82]: you can use a different variable name for err e.g. ferr

File: pipeline/runtime/runtest.go [hunk 5]
```diff
@@ -67,10 +67,10 @@ func executeRunTestStep(ctx context.Context, engine *engine.Engine, r *api.Start
...
 	if len(r.OutputVars) > 0 {
 		if exited != nil && exited.Exited && exited.ExitCode == 0 {
-			outputs := fetchExportedVarsFromEnvFile(outputFile, out)
+			outputs, err := fetchExportedVarsFromEnvFile(outputFile, out) //nolint:govet
```
Comment (shubham149) [inline pipeline/runtime/runtest.go:73]: Do we have to add this code here
if len(r.OutputVars) > 0 {
			// only return err when output vars are expected
			return exited, outputs, exportEnvs, err
		}

File: pipeline/runtime/common.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (raghavharness) [reply pipeline/runtime/common.go:82]: okay

File: pipeline/runtime/runtest.go (hunk 5 above)
Comment (raghavharness) [reply pipeline/runtime/runtest.go:73]: If I don't add it then it will fail for plugins. Since plugin can or cannot export output vars.

Comment (raghavharness) [reply pipeline/runtime/runtest.go:73]: So only failing when len(r.OutputVars) > 0 but we got error while parsing outputs.

Comment (raghavharness) [reply pipeline/runtime/runtest.go:73]: So if you asked for runtest step, we already have a check `if len(r.OutputVars) > 0` here.

**Review by shubham149 — APPROVED** [review]

**Pull request merged**